addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/ReplacerParamUnitTest.java

```diff
@@ -0,0 +1,182 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.replacer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/** Unit test for {@link ReplacerParam}. */
+class ReplacerParamUnitTest {
+
+    private static final ReplacerParamRule REPORT_TO_RULE =
+            new ReplacerParamRule(
+                    ReplacerParam.REPORT_TO_DESC,
+                    MatchType.RESP_HEADER_STR,
+                    ReplacerParam.REPORT_TO_REGEX,
+                    true,
+                    ReplacerParam.REPORT_TO_REPLACEMENT,
+                    null,
+                    false);
+    private ReplacerParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new ReplacerParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("replacer[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty("replacer.confirmRemoveToken", state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveExpectedVersionInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getCurrentVersion(), is(equalTo(1)));
+    }
+
+    @Test
+    void shouldHaveExpectedNumberOfRulesInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getRules().size(), is(equalTo(4)));
+        ReplacerParamRule newRule = param.getRule(ReplacerParam.REPORT_TO_DESC);
+        assertNotNull(newRule);
+        assertThat(newRule.getDescription(), is(equalTo(ReplacerParam.REPORT_TO_DESC)));
```

Comment:
It should assert all, and "tomorrow" it's not new.

Reply:
cspRule? v1NewRule?

Reply:
Asserting all (in a not messy way) would require changing how the existing rules are added/handled. (To have constant details and be re-usable, etc).

Maybe like the name of the method says I should just check number here.

Reply:
Count does not check what's being added, say you added the wrong rule.

Reply:
So you want me to create constants for all the existing rules too and re-work how they're created/added?

Reply:
There's not much to maintain, those rules/values have not been touched in years (a quick look says at least 7).

Reply:
Okay